### DISCUSSION TITLE
dynamic creation of dags

### DISCUSSION THREAD

**Discussion by Cornelius-Liepelt** [apache/airflow#28696] [Q&A]
Hey everyone, my team and i are currently trying to realize the dynamic generation of Dags based on Data written in an SQL-Database. Our plan is to fill tables with data according to there dependencies to other tables. Currently we fetch all tables for an specific Dag from the Database and iterate through them creating the task-tree while we do that.
We realized that this leads to longer DAG parsing-times. This would be no problem in general but because the dag is processed everytime an task is run this actually made a big difference for us. We already implemented the experimental feature to only parse the tasks of the dag that is currently run but it still takes to long in our oppinion.
As I understand there are already provided functions and workflows for the dynamic generation of tasks. Would these be better suited? Also are these function able to generate dynamic task dependencies? Because as I see it these functions are only able to create tasks running in parallel.
Any input or ideas on your end to lower the parsing time would be much appreciated.

**Comment by potiuk** [comment]
Just read this chapter in our Best Practices docs: https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#dynamic-dag-generation. It leads to Actually - that chapter has link to https://airflow.apache.org/docs/apache-airflow/stable/howto/dynamic-dag-generation.html whi that has all the answers for you.
